src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -743,6 +745,14 @@ public static bool TryCompleteSendTo(SafeSocketHandle socket, ReadOnlySpan<byte>
                     errorCode = SocketError.Success;
                     return true;
                 }
+
+                if (socket.SendTimeout > 0 && (Environment.TickCount64 - start) >= socket.SendTimeout)
+                {
+                    // Timeout is set and reached it while trying to send.
+                    errorCode = SocketError.TimedOut;
+                    return false;
```

Comment:
I think you want to return true here, since we return true for other error cases.

Reply:
fixed.